course_inventory/published_date.py

```diff
@@ -33,32 +39,49 @@ def get_next_page_url(self, response):
             logging.debug('The api call do not have Link headers')
             return None
 
+        course_id = int(response.result().url.split('?')[0].split('/')[-1])
```

Comment:
Do you think there is any performance cost to calling `result` twice (once here, once above)? Also, this seems a little fragile, endpoint-specific and not very generaliz-able. Is there another way we can get the `course_id`? Could it be passed as a parameter?

Reply:
I guess you're also using this pattern below. If you don't want to change this, that's fine, just thought I would put this idea out there.

Reply:
I took `response.result()` as a variable,  I don't know how much overhead it will add for calling over and over. But this seems to be a good idea to do.
Since the response is Async I can't pass the course id and the best way to get the course_id is from the Response object. I have tested the pattern of URL that is sent with both regular and paginated URL seems to work

Reply:
Yeah, I think it makes sense to do what you did and only call `result()` once. Thanks